### PR TITLE
Fixing bad evaluation of `null` as an array key in the asserter

### PR DESCRIPTION
**Pull request by Grummfy**
example that will be fixed https://gist.github.com/Grummfy/6d267855f90f89f891e7

### CODE REVIEW

**Comment by Hywan** [comment]
@Grummfy Hello!

Thanks for the PR. Can you provide a test please? Just create `case_github_70` in this file: https://github.com/hoaproject/Ruler/blob/master/Test/Unit/Issue.php#L52-L66.

Your actual patch is correct and I am validating it. Thank you :smile:.

**Comment by Grummfy** [comment]
here we are

File: Test/Unit/Issue.php [hunk 1]
```diff
@@ -64,4 +64,22 @@ public function case_github_50()
...
+    
+    public function case_github_70()
+    {
+        $this
+            ->given(
+                $ruler               = new LUT(),
+                $rule                = 'variable["foo"] is null',
+                $context             = new LUT\Context(),
+                $context['variable'] = [
+					'foo'	=> null
```
Comment (Hywan) [inline Test/Unit/Issue.php]: Please, fix indentation :-).

File: Test/Unit/Issue.php [hunk 2]
```diff
@@ -64,4 +64,22 @@ public function case_github_50()
...
+                ]
+            )
+            ->when(function () use ($ruler, $rule, $context) {
+                $this->boolean($ruler->assert($rule, $context))->isTrue();
```
Comment (Hywan) [inline Test/Unit/Issue.php]: Why `$this->boolean(…)->isTrue()`? And why the closure?

We are trying to specify the expected behavior, so:
```php
->when($result = $ruler->assert($rule, $context))
->then
    ->boolean($result)
        ->isTrue();
```
This is the expected oracle I guess.

File: Test/Unit/Issue.php (hunk 1 above)
Comment (Hywan) [inline Test/Unit/Issue.php]: Even an line array would be perfect: `$context['variable'] = ['foo' => null]`

**Comment by Grummfy** [comment]
;)

**Comment by Hywan** [comment]
Merged and rebased into https://github.com/hoaproject/Ruler/commit/4610112e689baf4e0682d3c4e3fac20f89606b23.

Thank you very much!

**Comment by Hywan** [comment]
This PR is included inside the [1.15.09.22 snapshot](https://github.com/hoaproject/Ruler/releases/tag/1.15.09.22).

**Pull request closed without merging**